R/cmapplot.R

```diff
@@ -17,73 +17,125 @@
+                          #        which can be done easily using `legend_bump`
+    legend_key_size = 14, # [t,f] Size of the legend key item.
+    leading_title = 1,    # [ ,f] Text leading for Title text
+    leading_caption = 1   # [ ,f] Text leading for Caption text
   )
-  cmapplot_globals$use_whitney = length(whitney_fonts$family) == 3
-
-  # Use Whitney if available
-  if (cmapplot_globals$use_whitney) {
-    # Add fonts to R
-    grDevices::windowsFonts(
-      `Whitney Medium` = grDevices::windowsFont("Whitney Medium"),
-      `Whitney Book` = grDevices::windowsFont("Whitney Book"),
-      `Whitney Semibold` = grDevices::windowsFont("Whitney Semibold")
-    )
-
-    # establish font variables
-    cmapplot_globals$font = list(
-      strong = list(family = "Whitney Semibold", face = "plain"),
-      regular = list(family="Whitney Medium", face="plain"),
-      light = list(family="Whitney Book", face="plain")
-    )
+)
 
-  # Otherwise, use Calibri
+## Use Whitney or Calibri if on Windows -- *must* be done with .onLoad()
+.onLoad <- function(...) {
+  if (.Platform$OS.type == "windows") {
+
+    # Check for Whitney
+    all_fonts <- sysfonts::font_files()
+    whitney_fonts <- all_fonts[all_fonts$family %in% c("Whitney Medium", "Whitney Book", "Whitney Semibold") & all_fonts$face=="Regular", ]
+    cmapplot_globals$use_whitney = length(whitney_fonts$family) == 3
+
+    # Use Whitney if available
+    if (cmapplot_globals$use_whitney) {
+      # Add fonts to R
+      grDevices::windowsFonts(
+        `Whitney Medium` = grDevices::windowsFont("Whitney Medium"),
+        `Whitney Book` = grDevices::windowsFont("Whitney Book"),
+        `Whitney Semibold` = grDevices::windowsFont("Whitney Semibold")
+      )
+
+      # Update font variables
+      cmapplot_globals$font <<- list(
+        strong = list(family = "Whitney Semibold", face="plain"),
+        regular = list(family="Whitney Medium", face="plain"),
+        light = list(family="Whitney Book", face="plain")
+      )
+
+    # Otherwise, use Calibri
+    } else {
+      packageStartupMessage(
+        "WARNING: Whitney is not installed on this PC, so CMAP theme will default to Calibri"
+      )
+      # Add fonts to R
+      grDevices::windowsFonts(
+        `Calibri` = grDevices::windowsFont("Calibri"),
+        `Calibri Light` = grDevices::windowsFont("Calibri Light")
+      )
+
+      # Update font variables
+      cmapplot_globals$font <<- list(
+        strong = list(family="Calibri", face="bold"),
+        regular = list(family="Calibri", face="plain"),
+        light = list(family="Calibri Light", face="plain")
+      )
+    }
+
+    # If non-Windows machine, stick to Arial
   } else {
-    message("WARNING: Whitney is not installed on this PC, so CMAP theme will default to Calibri")
-    # Add fonts to R
-    grDevices::windowsFonts(
-      `Calibri` = grDevices::windowsFont("Calibri"),
-      `Calibri Light` = grDevices::windowsFont("Calibri Light")
```

Comment:
Do Calibri and Calibri light exist on every windows machine? Just wondering if it would be helpful to verify, now that Arial is the package's universal default.

Reply:
[Yes, back through Windows Vista.](https://docs.microsoft.com/en-us/typography/font-list/calibri#products-that-supply-this-font)